src/app/components/layout/side-menu/side-menu.component.ts

```diff
@@ -138,15 +145,17 @@ export class SideMenuComponent implements OnInit {
       return;
     }
     const menuOption = document.getElementById(menuLink + '-submenu') as HTMLDivElement;
+    // delay the submenu close by 350ms to avoid flickering
     setTimeout(function () {
       // If mouse is over the menu or child sub-menu then return and don't close the sub-menu   
       if (menuOption.matches(':hover')) {
         return;
       }
+      if (menuOption.classList.contains('show')) {
+        this.toggleSubmenuState(menuLink, false);
+      }
       menuOption.classList.remove('show');
-
-      this.toggleSubmenuState(menuLink, false);
-    }.bind(this), 200);
+    }.bind(this), 300);
```

Comment:
actually you can next to it as 

bind(this), 300); // delay the submenu close to avoid flickering